src/main/java/com/logistic/project/model/Warehouse.java

```diff
@@ -1,4 +1,16 @@
 package com.logistic.project.model;
 
+import lombok.Data;
+
+import java.util.Map;
+
+@Data
 public class Warehouse {
+    private String name;
+    private Map<Resource, Integer> resourceQuantity;
+
+    public Warehouse(String name, Map<Resource, Integer> resourceQuantity) {
```

Comment:
`@RequiredArgsConstructor`

Reply:
@Data is a convenient shortcut annotation that bundles the features of @ToString, @EqualsAndHashCode, @Getter / @Setter and @RequiredArgsConstructor together: In other words, @Data generates all the boilerplate that is normally associated with simple POJOs (Plain Old Java Objects) and beans: getters for all fields, setters for all non-final fields, and appropriate toString, equals and hashCode implementations that involve the fields of the class, and a constructor that initializes all final fields, as well as all non-final fields with no initializer that have been marked with @NonNull, in order to ensure the field is never null.